common/src/main/java/com/lightstep/tracer/shared/Util.java

```diff
@@ -27,4 +28,15 @@ protected Random initialValue() {
                             (long) (1024 * Math.random()));
         }
     };
+
+    static long protoTimeToEpochMicros(Timestamp timestamp) {
+        return timestamp.getSeconds() * 1000 * 1000 + timestamp.getNanos() / 1000;
+    }
+
+    static Timestamp epochTimeMicrosToProtoTime(long micros) {
+        Timestamp.Builder builder = Timestamp.newBuilder();
+        builder.setSeconds(micros / 1000 / 1000);
```

Comment:
pedantic nit: can you either add some parens or just write `1000000` here? I got burned by an operator precedence bug once and told myself "never again" :)